client/src/services/api.ts

```diff
@@ -120,14 +120,20 @@ function oidcFactory() {
     };
 }
 
+let isApiRequestInProgress = false;
```

Comment:
I think this needs to go inside the `metrics` function. Many pages try to load multiple metrics at the same time. I think the current implementation would only allow one at a time

Reply:
Yes, good catch.